### PR TITLE
Set up in-context signup for Android

### PR DESCRIPTION
**Pull request by alistairjcbrown**
**Reviewer:** @shakyShane @GioSensation 
**Asana:** https://app.asana.com/0/0/1205688727602722/f

## Description
Adds necessary changes to support in-context signup in Android

## Steps to test
See latest test builds -- https://app.asana.com/0/0/1205404319005943/f

### CODE REVIEW

File: src/deviceApiCalls/api.json [hunk 1]
```diff
@@ -203,6 +203,14 @@
     "closeEmailProtectionTab": {
       "type": "object",
       "description": "Used by macOS to close the Email Protection tab after successful in-context sign-up or login"
+    },
+    "ShowInContextEmailProtectionSignupPrompt": {
+      "type": "object",
+      "description": "Used by Android to open a the in-context signup prompt and report back when completed",
```
Comment (GioSensation) [inline src/deviceApiCalls/api.json:209]: Typo.

```suggestion
      "description": "Used by Android to open the in-context signup prompt and report back when completed",
```

**Review by GioSensation — APPROVED** [review]
This is looking good and I see from Asana that it's been tested thoroughly. Is there any flow we could/should capture in tests?

**Comment by GioSensation** [comment]
I've added this [one-liner](https://github.com/duckduckgo/duckduckgo-autofill/pull/397/files#diff-dbc70eadf3d33686dbf9289cb9cb8c7f6c9e033342e3cc847e58d7f281095c27R27) that is also useful to Android in another context. And fixed that typo 😉.

**Comment by alistairjcbrown** [comment]
@GioSensation I've added an integration test which confirm the initial dismissal check and the prompt message being sent when dax is clicked. Most of the flow happens on native UI, so there's not much to test here.

**Comment by GioSensation** [comment]
Lovely. Ship it. 🚀

**Pull request merged**